classes/class.ilShortLinkGUI.php

```diff
@@ -0,0 +1,360 @@
+require_once('Customizing/global/plugins/Services/UIComponent/UserInterfaceHook/ShortLink/classes/class.ilShortLinkAccess.php');
+include_once('Customizing/global/plugins/Services/UIComponent/UserInterfaceHook/ShortLink/classes/class.ilShortLinkTableGUI.php');
+require_once('Services/Utilities/classes/class.ilConfirmationGUI.php');
+include_once('Services/Form/classes/class.ilPropertyFormGUI.php');
+require_once('Services/Repository/classes/class.ilObjectPluginGUI.php');
+
+
+/**
+ * GUI-Class ilShortLinkGUI
+ * This class manages the ShortLink frontend.
+ *
+ * @author              Thomas Kolonko <thomas.kolonko@ilub.unibe.ch>
+ * @version             $Id$
+ *
+ * @ilCtrl_isCalledBy   ilShortLinkGUI: ilUIPluginRouterGUI
+ */
+class ilShortLinkGUI extends ilObjectPluginGUI {
+    /**
+     * @var ilTemplate $my_tpl
+     */
+    protected $my_tpl;
+    /**
+     * @var ilCtrl $ctrl
+     */
+    protected $ctrl;
+    /**
+     * @var ilToolbarGUI $toolbar
+     */
+    protected $toolbar;
+    /**
+     * @var ilObjShortLink $obj
+     */
+    protected $obj;
+    /**
+     * @var ilShortLinkAccess $shortLinkAccessChecker
+     */
+    protected $shortLinkAccessChecker;
+    /**
+     * @var ilPropertyFormGUI $form
+     */
+    protected $form;
+    /**
+     * @var ilObjShortLink $externalFeedBlock
+     */
+    protected $externalFeedBlock;
+
+    public function __construct() {
+        global $ilCtrl, $tpl;
+        $this->ctrl = $ilCtrl;
+        $this->form = new ilPropertyFormGUI();
+
+        $this->shortLinkAccessChecker = new ilShortLinkAccess();
+
+        $this->obj = new ilObjShortLink();
+
+        $this->my_tpl = $tpl;
+        $this->my_tpl->getStandardTemplate();
+
+        $this->pl = new ilShortLinkPlugin();
+    }
+
+    /**
+     * @return bool
+     */
+    public function executeCommand() {
+        $cmd = $this->ctrl->getCmd();
+
+        switch($cmd){
+            case 'add':
+            case 'save':
+                $this->$cmd();
+                break;
+            case 'edit':
+            case 'delete':
+            case 'confirmedDelete':
+            case 'doUpdate':
+                if($_GET['link_id'] != NULL) {
+                    $this->shortLinkAccessChecker->checkPermission($this->obj, $_GET['link_id']);
+                } else if($_POST['obj_id'] != NULL) {
+                    $this->shortLinkAccessChecker->checkPermission($this->obj, $_POST['obj_id']);
+                } else if($_POST['shortLink_id'] != NULL) {
+                    $this->shortLinkAccessChecker->checkPermission($this->obj, $_POST['shortLink_id']);
+                } else {
+                    ilUtil::sendFailure($this->pl->txt("mapping_wrong"), true);
+                    ilUtil::redirect('login.php?baseClass=ilPersonalDesktopGUI');
+                }
+                $this->$cmd();
+                break;
+            case 'listShortLinks':
+                $this->$cmd();
```

Comment:
the checks are within the functions that list the entries. I guess it would be cleaner to just remove the ShortLink Button from the NavBar in that case.